Tribler/Main/Build/Win/tribler.nsi

```diff
@@ -325,72 +322,39 @@ SectionEnd
-	Quit
-${EndSwitch}
-
-SetShellVarContext all
-!macroend
-
-
 Function .onInit
-  !insertmacro Init "installer"
-
-  System::Call 'kernel32::CreateMutexA(i 0, i 0, t "Tribler") i .r1 ?e'
-
-  Pop $R0
-
-  StrCmp $R0 0 +3
+    System::Call 'kernel32::CreateMutexA(i 0, i 0, t "Tribler") i .r1 ?e'
 
-  MessageBox MB_OK "The installer is already running."
-  Abort
+    Pop $R0
+    StrCmp $R0 0 checkinst
 
-  ReadRegStr $R0 HKLM "SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\${PRODUCT}" "UninstallString"
-  StrCmp $R0 "" done
-  IfFileExists $R0 +1 done
-
-  MessageBox MB_OKCANCEL|MB_ICONEXCLAMATION "${PRODUCT} is already installed. $\n$\nClick `OK` to remove the previous version or `Cancel` to cancel this upgrade." /SD IDCANCEL IDOK uninst
-  Abort
-
-  uninst:
-    ClearErrors
-    ; Laurens (2016-03-29): Retrieve the uninstallString stored in the register. Do NOT use $INSTDIR as this points to the current $INSTDIR var of the INSTALLER, 
-    ; which is the default location at this point.
-    ReadRegStr $R0 HKLM "Software\Microsoft\Windows\CurrentVersion\Uninstall\${PRODUCT}" "UninstallString"
-    ExecWait '"$R0"' ;Do not copy the uninstaller to a temp file
-    ReadRegStr $R0 HKLM "SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\${PRODUCT}" "UninstallString"
-    StrCmp $R0 "" done
+    MessageBox MB_OK "The installer is already running."
     Abort
-  done:
 
-FunctionEnd
+    checkinst:
+        ReadRegStr $R0 HKLM "SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\${PRODUCT}" "UninstallString"
+        StrCmp $R0 "" done
+        IfFileExists $R0 showuninstdialog done
+
+    showuninstdialog:
+        MessageBox MB_OKCANCEL|MB_ICONEXCLAMATION "${PRODUCT} is already installed. $\n$\nClick `OK` to remove the previous version or `Cancel` to cancel this upgrade." /SD IDCANCEL IDOK uninst
+        Abort
+
+    uninst:
+        ClearErrors
+        ; Laurens (2016-03-29): Retrieve the uninstallString stored in the register. Do NOT use $INSTDIR as this points to the current $INSTDIR var of the INSTALLER, 
+        ; which is the default location at this point.
+        ReadRegStr $R0 HKLM "Software\Microsoft\Windows\CurrentVersion\Uninstall\${PRODUCT}" "UninstallString"
+        ExecWait '"$R0" _?=$INSTDIR' ; This prevents the installer from being ran in a tmp directory, causing execwait not to wait.
+        ReadRegStr $R0 HKLM "SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\${PRODUCT}" "UninstallString"
+        StrCmp $R0 "" done
+        Abort
+    done:
 
-Function un.onInit
-	SetShellVarContext all
-    	!insertmacro Init "uninstaller"
 FunctionEnd
 
 Function PageFinishRun
-	!insertmacro UAC_AsUser_ExecShell "" "$INSTDIR\tribler.exe" "" "" ""
+    ; This function is invoked if the user leaves the checkbox checked at the last page (do you wish to run...)
+    ; It calls the freshly created desktop shortcut
+    ExecShell "" "$DESKTOP\${PRODUCT}.lnk"
```

Comment:
Is a desktop shortcut always created? Is it possible to run `tribler.exe` immediately from the installed directory?